### DISCUSSION TITLE
Grain growth model with inclination dependence

### DISCUSSION THREAD

**Discussion by oarcelus** [idaholab/moose#24134] [Q&A Modules: Phase field] [answered]
Hi I am new to MOOSE and I would like to use it for developin a multi-phase field model for grain growth with inclination dependence. I would like to basically reproduce the results shown in this paper (https://doi.org/10.1016/j.actamat.2020.02.043). I have read the documentation quite extensively and followed some tutorials and examples, but I find that many of the parts regarding multi-phase field models are still not totally documented and I find myself lost in trying to implement the model.

Could I get some basic guidance into trying to implement this? I am particularly interested in including the grain-boundary energy terms shown into the linked paper (Seciton 2.2), into the Free-Energy expression.

**Comment by lindsayad** [comment]
@dschwen @permcody any thoughts?

**Reply by GiudGiud** [reply]
@jiangwen84 @laagesen

**Comment by laagesen** [comment, marked as answer]
Hi @oarcelus, thanks for your interest in the MOOSE framework! As you noted, we do not have much implemented in the phase-field module for multi-phase field models such as the one you linked to. I wonder if you have considered a multi-order parameter approach instead? We do have a lot more implemented in MOOSE using a multi-order parameter strategy, primarily using this model:

https://journals.aps.org/prb/abstract/10.1103/PhysRevB.78.024113

This model can include misorientation and inclination dependence. So, I think that either a multi-phase field or multi-order parameter approach could simulate the physics you are interested in; there would just be a lot more already implemented if you chose the multi-order parameter approach. There might still be some code that would need to be implemented depending on what inclination-dependent features you want to add, there would just be a lot more you could build on.

**Comment by oarcelus** [comment]
Hi @laagesen, thank you for your comment. In fact, I read the paper you just linked, as well as the last one by N. Moelans (https://www.sciencedirect.com/science/article/pii/S0264127522002131?via%3Dihub) for which I found some MOOSE implementation (https://github.com/PengWei97/yinglong), which apparently also solves for some known [issue](https://github.com/idaholab/moose/issues/13608) regarding the use of GrainTracker and GBAnisotropy? (not sure of this latter one).

In any case, I am still a bit confused about all this digging (jumping fresh into this giant software can be confusing...). So I will post a few more questions here:

1) I fail to understand the difference between 'multi-phase' and 'multi-order parameter'. In which way the paper I originally linked is considered 'multi-phase' and the ones by N.Moelans are not? They both state some sort of free energy and they solve for the time evolution of non-conserved parameters through a first derivative of time and a functional derivative of the free-energy with respect to the order parameters. The only difference I see is that in the paper I linked there are some extra-higher order terms, which will probably be a nightmare to treat in code, but other than that nothing more.

2) The 'GrainTracker' functionallity is 'in principle' only used to improve the computational performance of the models right? You could in principle make a grain growth model considering explicitly all of the order parameters, then a single Kernel would take care of all the interactions, since they are the same. What the GrainTracker does is to make it so that no two order parameters are contiguous (so that tails do not overlap), and you can reduce their number.  If so, is there any functionallity to merge grains as there is in GrainTracker, but without actually using GrainTracker?

3) In the link I originally sent there is some pretty straightforward formulas linking the grain-boundary energy to the grain inclination, making it a continuous function. Is there a way of implementing this in a simple way? I guess that I also have an understanding gap between the terms 'misorientation' and 'inclination'.

Sorry for the long post, but these conversations are really useful to me, and thanks again for the response!

**Reply by laagesen** [reply]
Let me try to answer questions by number:

1. Multi-phase field models usually refers to models where the phase field represents the phase fraction and the phase-fields are required to sum to 1 at every position. Multi-order parameter models do not enforce such a restriction. Some people feel that in the multi-phase field models, the direct correspondence between the phase field variables and phase fraction makes the model more physical, but I think that if you are interested in simulating grain growth, it is more important that the Allen-Cahn evolution equations approach the evolution equations for grain growth in the sharp interface limit (for grain growth, this is that grain boundary velocity is proportional to local curvature). This has been done in the Moelans paper I referred to, I am sure it has also been done for multi-phase field models but I don't know a good reference for that off-hand. Several years ago there was a paper comparing multi-phase field and multi-order parameter approaches for simulating grain growth, I believe Steinbach and Moelans were among the co-authors, so if you do a search of Web of Science you can probably track that down (sorry I don't have the reference handy off hand here either).
2. You are right on the purpose of the GrainTracker. The only thing I would say is I am not sure what you mean by "a single Kernel would take care of all the interactions" - a Kernel in MOOSE parlance represents a single term in an equation. So if you had Allen-Cahn equations for every order parameter, you would need additional kernels to represent each term in each Allen-Cahn equation. This may be confusing because adding Kernels is hidden in the grain growth example input files because we use Actions, namely the PolycrystalKernelAction, which automates adding kernels to the input files.
3. Misorientation vs. inclination has to do with the 5 degrees of freedom of a grain boundary, which you may want to refer to a materials science textbook on. I believe misorientation refers to the 3 possible ways (angles) of orienting the crystals relative to each other, while inclination refers to the plane that cuts the two crystals which I think are two. If you can post an example of exactly what type of function you are wanting to implement, I can try to give some more specifics on how it could be implemented or if we have something similar already implemented.

**Reply by ykvishal** [reply]
Just to want to give the reference mentioned in point 1 
Moelans, Nele, Frank Wendler, and Britta Nestler. "Comparative study of two phase-field models for grain growth." Computational Materials Science 46.2 (2009): 479-490. [https://www.sciencedirect.com/science/article/abs/pii/S092702560900158X](https://www.sciencedirect.com/science/article/abs/pii/S092702560900158X)

**Reply by laagesen** [reply]
thank you!

**Reply by laagesen** [reply]
especially since I was incorrect on the co-authors! :-)
